include/fuzzuf/utils/vfs/read_once.hpp

```diff
@@ -0,0 +1,88 @@
+/*
+ * fuzzuf
+ * Copyright (C) 2022 Ricerca Security
+ * 
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+/**
+ * @file map_file.hpp
+ * @author Ricerca Security <fuzzuf-dev@ricsec.co.jp>
+ */
+#ifndef FUZZUF_INCLUDE_UTILS_VFS_READ_ONCE_HPP
+#define FUZZUF_INCLUDE_UTILS_VFS_READ_ONCE_HPP
+#include <fcntl.h>
+#include "fuzzuf/utils/filesystem.hpp"
+#include "fuzzuf/utils/map_file.hpp"
+
+namespace fuzzuf::utils::vfs::adaptor {
+
+template< typename Base >
```

Comment:
Please describe what kind of types can be passed as `Base`. 
Since the precise conditions are too long to write, some simple explanations like "Basically, `Base` should be a type that derives from `VFS`. Additionally, it must have some extra member functions like `OpenDirectoryRecursive`. For example, LocalFilesystem meets these conditions." are enough.

Reply:
fixed